core/core.go

```diff
@@ -552,6 +552,40 @@ func (app *App) Start(ctx context.Context) error {
 		orderWatcherErrChan <- app.orderWatcher.Watch(innerCtx)
 	}()
 
+	// Ensure that RPC client is on the same ChainID as is configured with ETHEREUM_CHAIN_ID
+	chainIdCheckerWg := &sync.WaitGroup{}
+	chainIdMismatchErrChan := make(chan error, 1)
+	wg.Add(1)
+	chainIdCheckerWg.Add(1)
+	go func() {
+		defer wg.Done()
+		defer chainIdCheckerWg.Done()
+		ctx, cancel := context.WithTimeout(context.Background(), ethereumRPCRequestTimeout)
```

Comment:
Can we move the actual logic in this go routine into a helper method with a good name? Having it in-lined here feels like the wrong level of abstraction when compared with the rest of it's parent's function logic.

Reply:
Use `innerCtx` instead of `context.Background()` here.